aziotd/src/main.rs

```diff
@@ -226,21 +228,50 @@ where
 
     log::info!("Starting server...");
 
-    let incoming = connector
+    let mut incoming = connector
         .incoming()
         .await
         .map_err(|err| ErrorKind::Service(Box::new(err)))?;
-    let server = hyper::Server::builder(incoming).serve(hyper::service::make_service_fn(|_| {
-        let server = server.clone();
-        async move { Ok::<_, std::convert::Infallible>(server.clone()) }
-    }));
-    let () = server
-        .await
-        .map_err(|err| ErrorKind::Service(Box::new(err)))?;
 
-    log::info!("Stopped server.");
-
-    Ok(())
+    match &mut incoming {
```

Comment:
If you keep this as a method of `http_common::Incoming` like:

```rust
async fn serve(&mut self, server: impl hyper::service::Service<...> + Clone + Send + 'static) -> std::io::Result<()> {
    match self { ... }
}
```

... (in place of the `hyper::Accept` impl that this replaces), you can avoid needing to make the `http_common::connector` and `::uid` modules `pub`.